File: 0070-climbing-stairs/0070-climbing-stairs.cpp
class Solution {
    int climbStairs(int n, vector<int>&dp) {
        if(n<=1)
            return 1;   
           
        //c(3) = c(2) + c(1);
        //c(4) =  c(3) + c(2);
        if(dp[n] != -1)
            return dp[n];
       return dp[n] = climbStairs(n-1, dp) + climbStairs(n-2,dp);  
        
    }
public:   
    int climbStairs(int n) {
      vector<int>dp(n+1, -1);  
     return climbStairs(n, dp);
    }
};

File: 0070-climbing-stairs/0070-climbing-stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        
        dp = {}
        
        def climb(n):
            
            if n <= 1:
                return 1
            if n in dp.keys():
                return dp[n]
            
            dp[n] = climb(n - 1) + climb(n - 2)
            return dp[n]
        return climb(n)
        